server/src/main/java/org/cloudfoundry/identity/uaa/codestore/JdbcExpiringCodeStore.java

```diff
@@ -121,7 +121,7 @@ public ExpiringCode peekCode(String code, String zoneId) {
 
         try {
             ExpiringCode expiringCode = jdbcTemplate.queryForObject(selectAllFields, rowMapper, code, zoneId);
-            if (expiringCode.getExpiresAt().getTime() < timeService.getCurrentTimeMillis()) {
+            if (expiringCode != null && expiringCode.getExpiresAt().getTime() < timeService.getCurrentTimeMillis()) {
```

Comment:
The refactoring changes are being tested by existing tests. Not sure if it is worthwhile to mock a null being returned by the query and then making sure we get a handled exception rather than a null pointer exception.